src/EFCore.PG/Properties/NpgsqlStrings.Designer.cs

```diff
@@ -1,254 +1,557 @@
+                    ref ((Diagnostics.Internal.NpgsqlLoggingDefinitions)logger.Definitions).LogExpressionIndexSkipped,
+                    () => new EventDefinition<string, string>(
+                        logger.Options,
+                        NpgsqlEventId.ExpressionIndexSkippedWarning,
+                        LogLevel.Warning,
+                        "NpgsqlEventId.ExpressionIndexSkippedWarning",
+                        level => LoggerMessage.Define<string, string>(
+                            level,
+                            NpgsqlEventId.ExpressionIndexSkippedWarning,
+                            _resourceManager.GetString("LogExpressionIndexSkipped"))));
+            }
+
+            return (EventDefinition<string, string>)definition;
+        }
+
 
         /// <summary>
+
         ///     Constraint '{name}' on table {tableName} cannot be scaffolded because it includes a column that cannot be scaffolded (e.g. enum).
+
         /// </summary>
-        public static readonly EventDefinition<string, string> LogUnsupportedColumnConstraintSkipped
-            = new EventDefinition<string, string>(
-                NpgsqlEventId.UnsupportedColumnConstraintSkippedWarning,
-                LogLevel.Warning,
-                LoggerMessage.Define<string, string>(
-                    LogLevel.Warning,
-                    NpgsqlEventId.UnsupportedColumnConstraintSkippedWarning,
-                    _resourceManager.GetString("LogUnsupportedColumnConstraintSkipped")));
+
+        public static EventDefinition<string, string> LogUnsupportedColumnConstraintSkipped([NotNull] IDiagnosticsLogger logger)
+        {
+            var definition = ((Diagnostics.Internal.NpgsqlLoggingDefinitions)logger.Definitions).LogUnsupportedColumnConstraintSkipped;
+            if (definition == null)
+            {
+                definition = LazyInitializer.EnsureInitialized<EventDefinitionBase>(
+                    ref ((Diagnostics.Internal.NpgsqlLoggingDefinitions)logger.Definitions).LogUnsupportedColumnConstraintSkipped,
+                    () => new EventDefinition<string, string>(
+                        logger.Options,
+                        NpgsqlEventId.UnsupportedColumnConstraintSkippedWarning,
+                        LogLevel.Warning,
+                        "NpgsqlEventId.UnsupportedColumnConstraintSkippedWarning",
+                        level => LoggerMessage.Define<string, string>(
+                            level,
+                            NpgsqlEventId.UnsupportedColumnConstraintSkippedWarning,
+                            _resourceManager.GetString("LogUnsupportedColumnConstraintSkipped"))));
+            }
+
+            return (EventDefinition<string, string>)definition;
+        }
+
 
         /// <summary>
+
         ///     Index '{name}' on table {tableName} cannot be scaffolded because it includes a column that cannot be scaffolded (e.g. enum).
+
```

Comment:
Why are there to many new lines? Where are they from?

Reply:
This is an auto-generated file - the template managing it is NpgsqlStrings.Designer.tt, and it has changed significantly... I wouldn't worry too much about it...